sdk/core/src/response.rs

```diff
@@ -58,7 +58,23 @@ impl Response {
     }
 
     pub fn deconstruct(self) -> (StatusCode, HeaderMap, PinnedStream) {
-        (self.status, self.headers, self.response)
+        (self.status, self.headers, self.body)
+    }
+
+    pub async fn validate(self, expected_status: StatusCode) -> Result<Self, crate::HttpError> {
+        let status = self.status();
+        if expected_status != status {
+            let body = collect_pinned_stream(self.body)
```

Comment:
I'm not sure what you mean with regards to logging. We're not logging anything here. Can you explain more?

Reply:
In this case, no, you're not logging but you're [including the body](https://github.com/Azure/azure-sdk-for-rust/pull/286/files/793dbc00ab24a853e4952ab8896db12e5d9933f8#diff-5e3e9406014c7c90e2c7dd3f7616bc517d15bfdc63b60e47b55eb7bdd7d2a024R73) in the error, which may get logged higher up the stack. We can't expose secret information so easily. Some apps may not even realize the body - potentially containing secrets - is in an error that gets logged higher up.

Reply:
Ok. I'll create an issue for this, but this is the current functionality in all endpoints and not something I changed so I don't think it should block this PR.

Reply:
I created an [issue](https://github.com/Azure/azure-sdk-for-rust/issues/289). I do have a concern that this makes debugging things harder since you presumably on errors the body is exposing some information of why the request failed. I'd love for you provide your insight on that issue.

Reply:
Security is foremost important. For methods with models, any errors serializing a body we should catch worry code coverage. Issues may still occur but we can fix those in debugging, which we can turn on content logging. Customers sending raw JSON payloads can do the same if they discover an issues. We just don't expose body current by default. This is critical.